File: money/franc.go
package money

type Franc struct {
	amount int
}

func newFranc(amount int) *Franc {
	d := new(Franc)
	d.amount = amount
	return d
}

func (d *Franc) times(multiplier int) *Franc {
	return newFranc(d.amount * multiplier)
}

func (d *Franc) equals(object *Franc) bool {
	return d.amount == object.amount
}

File: python/franc.py
from typing import Any

from money import Money


class Franc(Money):
    def __init__(self, amount: int, currency: str) -> None:
        super().__init__(amount, currency)

    def times(self, multiplier: int) -> Money:
        """
        This method is [exprain]
        docstringの書き方に派閥があるらしい(googleとか
        """
        return Money.franc(self.amount * multiplier)
